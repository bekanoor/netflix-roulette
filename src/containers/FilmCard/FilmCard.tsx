import React from 'react'

type stateType = {
  page: string
  movieId: string
}

interface IProps {
  cover: string
  filmTitle: string
  releaseDate: string
  genre: string
  id: string
  onChangePage: (value: stateType) => void
}

const FilmCard = (props: IProps) => {
  const { onChangePage, cover, filmTitle, genre, releaseDate, id} = props

  const handelChangePage = (e: React.MouseEvent<HTMLDivElement>) => {
    const id = e.currentTarget.id
    if (onChangePage) onChangePage({ page: 'view', movieId: id })
  }

  return (
    <div className='film-card' onClick={handelChangePage} id={id}>
      <div className='film-card__image-container'>
        <img className='film-card__image' src={cover} alt='image'></img>
      </div>
      <div className='film-card__title-container'>
        <p className='film-card__title'>{filmTitle}</p>
        <button className='film-card__release-date'>{releaseDate}</button>
      </div>
      <p className='film-card__genre'>{genre}</p>
    </div>
  )
}

export { FilmCard }
