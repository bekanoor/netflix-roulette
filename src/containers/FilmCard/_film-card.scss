.film-card {
  width: 320px;
  cursor: pointer;
  margin: 25px 0;
  
  &__title {
    text-transform: uppercase;
    font-weight: 500;
    font-size: 18px;
    line-height: 22px;
    color: #FFFFFF;
    mix-blend-mode: normal;
    opacity: 0.7;

  }
  &__image-container {
  height: 450px;
  }
  &__image {
    width: 100%;
    height: 100%;
  }
  &__title-container {
    margin-top: 10px;
    display: flex;
    justify-content: space-between;
  }
  &__release-date {
    color: white;
    font-family: 'Montserrat', sans-serif;
    font-size: 14px;
    width: 66px;
    height: 26px;
    border: 1px solid rgba($color: #ffffff, $alpha: 0.3);
    background-color: #232323;
    border-radius: 5px;
  }
  &__genre {
    font-weight: 500;
    font-size: 14px;
    line-height: 17px;
    color: #FFFFFF;
    mix-blend-mode: normal;
    opacity: 0.5;
  }
}