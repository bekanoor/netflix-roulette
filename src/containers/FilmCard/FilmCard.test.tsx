import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import { FilmCard } from './FilmCard'

describe('test film card', () => {
  test('should render film card correctly', () => {
    const { container } = render(
      <FilmCard
        cover=''
        filmTitle=''
        releaseDate={1}
        genre=''
        id=''
        onChangePage={() => {}}
      />
    )

    expect(container).toMatchSnapshot()
  })

  test('dom elements really exist', () => {
    render(
      <FilmCard
        cover=''
        filmTitle=''
        releaseDate={1}
        genre=''
        id=''
        onChangePage={() => {}}
      />
    )

    const card = screen.getByTestId('film-card-test')
    expect(card).toBeInTheDocument
  })
})
