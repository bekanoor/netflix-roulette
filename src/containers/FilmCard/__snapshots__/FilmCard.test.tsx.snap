// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test film card should render film card correctly 1`] = `
<div>
  <div
    class="film-card"
    data-testid="film-card-test"
    id=""
  >
    <div
      class="film-card__image-container"
    >
      <img
        alt="image"
        class="film-card__image"
        src=""
      />
    </div>
    <div
      class="film-card__title-container"
    >
      <p
        class="film-card__title"
      />
      <button
        class="film-card__release-date"
      >
        1
      </button>
    </div>
    <p
      class="film-card__genre"
    />
  </div>
</div>
`;
